gallery/visualtests/nxDrawer.js

```diff
@@ -0,0 +1,189 @@
+        await openDrawer(buttonId, drawerId);
+        await a11yTest();
+      });
+    });
+  });
+
+  describe('NxDrawer with Narrow Variant', function() {
+    const browserSetup = setupBrowser('#/NxDrawerVariantExample', false);
+    const { getPage, checkFullPageScreenshot, a11yTest } = browserSetup;
+    const openDrawer = openDrawerFromPage(browserSetup);
+
+    const buttonId = 'nx-drawer-variant-narrow-open-button';
+    const drawerId = 'nx-drawer-variant-narrow';
+
+    it('looks right', async function() {
+      await getPage().setViewport(viewportSize);
+      await openDrawer(buttonId, drawerId);
+      await checkFullPageScreenshot();
+    });
+
+    it('passes a11y', async function() {
+      await getPage().setViewport(viewportSize);
+      await openDrawer(buttonId, drawerId);
+      await a11yTest();
+    });
+  });
+
+  describe('NxDrawer with Overflowing Form', function() {
+    const browserSetup = setupBrowser('#/NxDrawerWithNxFormExample', false);
+    const { getPage, checkFullPageScreenshot, a11yTest } = browserSetup;
+    const openDrawer = openDrawerFromPage(browserSetup);
+
+    const buttonId = 'nx-drawer-with-nx-form-overflowing-open-button';
+    const drawerId = 'nx-drawer-with-nx-form-overflowing';
+
+    it('looks right', async function() {
+      await getPage().setViewport(viewportSize);
+      await openDrawer(buttonId, drawerId);
+      await checkFullPageScreenshot();
+    });
+
+    it('passes a11y', async function() {
+      await getPage().setViewport(viewportSize);
+      await openDrawer(buttonId, drawerId);
+      await a11yTest();
+    });
+  });
+
+  describe('NxDrawer + NxDropdown ESC Closing Behavior', function() {
+    const { getPage, isFocused, isInDocument, waitAndGetElements, wait } = setupBrowser('#/NxDrawerEscExample', false);
+
+    async function pressEsc() {
+      const { keyboard } = getPage();
+      await keyboard.press('Escape');
+    }
+
+    it('closes one layer per ESC press', async function() {
+      const [openButton] = await waitAndGetElements('#nx-drawer-esc-open-button');
+
+      await openButton.click();
+
+      await wait(200);
+
+      const [drawer, dropdownToggle] = await waitAndGetElements(
+          '#nx-drawer-esc',
+          '#nx-drawer-esc .nx-dropdown__toggle'
+      );
+
+      expect(await isInDocument(drawer)).toBe(true);
+
+      expect(await isInDocument(dropdownToggle)).toBe(true);
+
+      await dropdownToggle.click();
+
+      const [dropdownMenu] = await waitAndGetElements('#nx-drawer-esc .nx-dropdown-menu');
+
+      expect(await isInDocument(dropdownMenu)).toBe(true);
+
+      await pressEsc();
+
+      expect(await isInDocument(drawer)).toBe(true);
+
+      expect(await isInDocument(dropdownMenu)).toBe(false);
+
+      expect(await isFocused(dropdownToggle)).toBe(true);
+
+      await pressEsc();
+
+      await wait(200);
+
+      expect(await isInDocument(drawer)).toBe(false);
+
+      expect(await isFocused(openButton)).toBe(true);
```

Comment:
Minor: blank lines between every statement is excessive.  Grouping similarly themed adjacent lines can make things easier to read.  In tests, this often comes out as one of two patterns:
1. Groups of setup/actions and groups of assertions
2. Groups that each contain the some setup/action and then the assertions about the result of that setup/action.

For this function, I might do it like this:
```suggestion
      const [openButton] = await waitAndGetElements('#nx-drawer-esc-open-button');

      await openButton.click();
      await wait(200);

      const [drawer, dropdownToggle] = await waitAndGetElements(
          '#nx-drawer-esc',
          '#nx-drawer-esc .nx-dropdown__toggle'
      );

      expect(await isInDocument(drawer)).toBe(true);
      expect(await isInDocument(dropdownToggle)).toBe(true);

      await dropdownToggle.click();

      const [dropdownMenu] = await waitAndGetElements('#nx-drawer-esc .nx-dropdown-menu');

      expect(await isInDocument(dropdownMenu)).toBe(true);

      await pressEsc();

      expect(await isInDocument(drawer)).toBe(true);
      expect(await isInDocument(dropdownMenu)).toBe(false);
      expect(await isFocused(dropdownToggle)).toBe(true);

      await pressEsc();
      await wait(200);

      expect(await isInDocument(drawer)).toBe(false);
      expect(await isFocused(openButton)).toBe(true);
```